### PR TITLE
Editing an agent returns you to correct scenario or index

### LINKED ISSUE #1002: Redirect back to scenario page
it would be better to it redirects back to scenario page if i edit agent from its scenario page

### PR DESCRIPTION
**Pull request by cantino**
Should fix #1002.

@knu, does this look good?

### CODE REVIEW

**Comment by cantino** [comment]
@knu, this refactor look okay?

**Comment by knu** [comment]
@cantino Looks good!

File: app/views/agents/_table.html.erb [hunk 1]
```diff
@@ -14,7 +14,7 @@
     <% @agents.each do |agent| %>
       <tr>
         <td class='<%= "agent-unavailable" if agent.unavailable? %>'>
-          <%= link_to agent.name, agent_path(agent) %>
+          <%= link_to agent.name, agent_path(agent, return: (defined?(return_to) && return_to) || request.path) %>
```
Comment (knu) [inline app/views/agents/_table.html.erb]: I just googled and read `defined?()` would work reliably for view locals in Rails 4.2, where you would have previously used `local_assigns.has_key?`.  Nice to know!

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -30,6 +30,20 @@ def upgrade_warning
     basecamp_auth_check
   end
 
+  def filtered_agent_return_link(options = {})
+    case ret = params[:return].presence || options[:return]
+      when "show"
```
Comment (knu) [inline app/controllers/application_controller.rb]: @cantino I'm not familiar with this indent style in ruby.  Is it your preference, or just that you wrote it without auto-indent?

Comment (cantino) [inline app/controllers/application_controller.rb]: This is what I usually use, although I don't feel strongly.  I find it easier to read if the `when` statements are indented one level from the `case`.

**Comment by cantino** [comment]
Thanks!

**Pull request merged**